File: PetShop.cpp
#include <iostream>
#include <string>

using namespace std;

class Product {
private:
    int id;
    string nama;
    string kategori;
    int harga;

public:
    Product() {}
    Product(int id, string nama, string kategori, int harga) {
        this->id = id;
        this->nama = nama;
        this->kategori = kategori;
        this->harga = harga;
    }

    int get_id() {
        return this->id;
    }

    void set_id(int id) {
        this->id = id;
    }

    string get_nama() {
        return this->nama;
    }

    void set_nama(string nama) {
        this->nama = nama;
    }

    string get_kategori() {
        return this->kategori;
    }

    void set_kategori(string kategori) {
        this->kategori = kategori;
    }

    int get_harga() {
        return this->harga;
    }

    void set_harga(int harga) {
        this->harga = harga;
    }

    ~Product()
    {
            
    }
};

class PetShop {
private:
    Product products[100];
    int productCount = 0;
    int nextId = 1;

public:
    void displayProducts();
    void addProduct(string nama, string kategori, int harga);
    void updateProduct(int id, string newnama, string newkategori, int newharga);
    void deleteProduct(int id);
    void searchProduct(string nama);
};

void PetShop::displayProducts() {
    if (productCount == 0) {
        cout << "Produk tidak ditemukan.\n";
        return;
    }
    cout << "\nList Produk:\n";
    for (int i = 0; i < productCount; i++) {
        cout << "ID: " << products[i].get_id() << ", nama: " << products[i].get_nama() << ", kategori: " << products[i].get_kategori() << ", harga: Rp " << products[i].get_harga() << endl;
    }
}

void PetShop::addProduct(string nama, string kategori, int harga) {
    if (productCount < 100) {
        products[productCount] = Product(nextId++, nama, kategori, harga);
        productCount++;
        cout << "Berhasil menambahkan produk.\n";
    } else {
        cout << "List Produk Penuh.\n";
    }
}

void PetShop::updateProduct(int id, string newnama, string newkategori, int newharga) {
    for (int i = 0; i < productCount; i++) {
        if (products[i].get_id() == id) {
            products[i].set_nama(newnama);
            products[i].set_kategori(newkategori);
            products[i].set_harga(newharga);
            cout << "Berhasil mengupdate produk.\n";
            return;
        }
    }
    cout << "Produk tidak ditemukan.\n";
}

void PetShop::deleteProduct(int id) {
    for (int i = 0; i < productCount; i++) {
        if (products[i].get_id() == id) {
            for (int j = i; j < productCount - 1; j++) {
                products[j] = products[j + 1];
            }
            productCount--;
            cout << "Berhasil menghapus produk.\n";
            return;
        }
    }
    cout << "Produk tidak ditemukan.\n";
}

void PetShop::searchProduct(string nama) {
    for (int i = 0; i < productCount; i++) {
        if (products[i].get_nama() == nama) {
            cout << "Found: ID: " << products[i].get_id() << ", nama: " << products[i].get_nama() << ", kategori: " << products[i].get_kategori() << ", harga: Rp " << products[i].get_harga() << endl;
            return;
        }
    }
    cout << "Produk tidak ditemukan.\n";
}

File: JAVA/Petshop.java
import java.util.ArrayList; 
import java.util.List;      

class PetShop {
    private List<String[]> products; 

    public PetShop() {
        products = new ArrayList<>(); 
    }

    // Metode untuk menampilkan semua produk
    public void displayProducts() {
        if (products.isEmpty()) { // Memeriksa apakah daftar produk kosong
            System.out.println("Tidak ada data produk.");
            return;
        }
        System.out.println("Daftar Produk:"); 
        for (String[] product : products) { 
            System.out.printf("Id: %s, Nama: %s, Kategori: %s, Harga: Rp %s%n",
                    product[0], product[1], product[2], product[3]); // Menampilkan informasi produk
        }
    }

    // Metode untuk menambahkan produk baru
    public void addProduct(String name, String category, int price) {
        int productId = products.size() + 1; // Menghasilkan ID produk baru
        String[] newProduct = {String.valueOf(productId), name, category, String.valueOf(price)}; // Membuat array produk baru
        products.add(newProduct); // Menambahkan produk ke daftar
        System.out.printf("Produk '%s' berhasil ditambahkan.%n", name); 
    }

    // Metode untuk memperbarui informasi produk
    public void updateProduct(int productId, String name, String category, Integer price) {
        for (String[] product : products) { 
            if (Integer.parseInt(product[0]) == productId) { // Memeriksa apakah ID produk cocok
                if (name != null) { // Memperbarui nama jika tidak null
                    product[1] = name;
                }
                if (category != null) { // Memperbarui kategori jika tidak null
                    product[2] = category;
                }
                if (price != null) { // Memperbarui harga jika tidak null
                    product[3] = String.valueOf(price);
                }
                System.out.printf("Produk dengan ID %d berhasil diperbarui.%n", productId); 
                return; 
            }
        }
        System.out.printf("Produk dengan ID %d tidak ditemukan.%n", productId); 
    }

    // Metode untuk menghapus produk berdasarkan ID
    public void deleteProduct(int productId) {
        for (String[] product : products) { 
            if (Integer.parseInt(product[0]) == productId) { // Memeriksa apakah ID produk cocok
                products.remove(product); // Menghapus produk dari daftar
                System.out.printf("Produk dengan ID %d berhasil dihapus.%n", productId); 
                return; 
            }
        }
        System.out.printf("Produk dengan ID %d tidak ditemukan.%n", productId); 
    }

    // Metode untuk mencari produk berdasarkan nama
    public void searchProduct(String name) {
        List<String[]> foundProducts = new ArrayList<>(); // Daftar untuk menyimpan produk yang ditemukan
        for (String[] product : products) { 
            if (product[1].toLowerCase().contains(name.toLowerCase())) { 
                foundProducts.add(product); // Menambahkan produk yang ditemukan ke daftar
            }
        }
        if (foundProducts.isEmpty()) { // Memeriksa apakah tidak ada produk yang ditemukan
            System.out.printf("Tidak ada produk yang ditemukan dengan nama '%s'.%n", name);
            return;
        }
        System.out.println("Produk yang ditemukan:"); 
        for (String[] product : foundProducts) { 
            System.out.printf("Id: %s, Nama: %s, Kategori: %s, Harga: Rp %s%n",
                    product[0], product[1], product[2], product[3]); // Menampilkan informasi produk yang ditemukan
        }
    }
}